homeassistant/components/homekit/type_covers.py

```diff
@@ -76,13 +83,25 @@ def set_state(self, value):
     def update_state(self, new_state):
         """Update cover state after state changed."""
         hass_state = new_state.state
-        if hass_state in (STATE_OPEN, STATE_CLOSED):
-            current_state = 0 if hass_state == STATE_OPEN else 1
-            self.char_current_state.set_value(current_state)
+        if hass_state in (STATE_OPEN, STATE_OPENING, STATE_CLOSED, STATE_CLOSING):
```

Comment:
Use the dictionary here as well:
```suggestion
        if hass_state in HASS_TO_HK_STATE:
```